docs/source/integrations/iceberg_versioning.md

```diff
@@ -0,0 +1,228 @@
+
+Create a temporary location for Iceberg tables by running the following command:
+
+```bash
+mkdir -p /tmp/warehouse
+```
+Then create or update a file called `.pyiceberg.yaml` in the root directory of the system with the following content:
+
+```yaml
+catalog:
+  default:
+    type: sql
+    uri: sqlite:////tmp/warehouse/pyiceberg_catalog.db
+    warehouse: file:///tmp/warehouse/warehouse
+```
+You can check if the configuration is by opening a Python shell with `ipython` command and running the following code:
+
+```python
+from pyiceberg.catalog import load_catalog
+catalog = load_catalog(name="default")
+```
+## Define a custom dataset to use Iceberg tables
+
+To use the Iceberg tables with Kedro, you will need to define a [custom dataset that uses the `pyiceberg` library](../data/how_to_create_a_custom_dataset.md). Create a new file  called `pyiceberg_dataset.py` in the `src/<package_name>` directory of your project and copy the following code:
+
+```python
+import pyarrow as pa
+from kedro.io.core import AbstractDataset, DatasetError
+from pyiceberg.catalog import load_catalog
+from pyiceberg.exceptions import NoSuchTableError
+
+DEFAULT_LOAD_ARGS = {"load_version": None}
+DEFAULT_SAVE_ARGS = {"mode": "overwrite"}
+
+class PyIcebergDataset(AbstractDataset):
+    def __init__(
+            self,
+            catalog,
+            namespace,
+            table_name,
+            table_type="pandas",
+            load_args=DEFAULT_LOAD_ARGS,
+            scan_args=None,
+            save_args=DEFAULT_SAVE_ARGS,
+    ):
+        self.table_name = table_name
+        self.namespace = namespace
+        self.catalog = load_catalog(catalog)
+        self.table_type = table_type
+        self.load_args = load_args
+        self.table = self._load_table(namespace, table_name)
+        self.save_args = save_args
+        self.scan_args = scan_args
+
+
+    def load(self):
+        if not self.table:
+            raise DatasetError("Table does not exist")
+        if self.scan_args:
+            scan = self.table.scan(**self.scan_args)
+        else:
+            scan = self.table.scan()
+        if self.table_type == "pandas":
+            return scan.to_pandas()
+        elif self.table_type == "arrow":
+            return scan.to_arrow()
+        elif self.table_type == "duckdb":
+            return scan.to_duckdb()
+        elif self.table_type == "daft":
+            return scan.to_daft()
+        else:
+            raise DatasetError("Table type not supported")
```

Comment:
Spark has good support for working with Iceberg, and a lot of Kedro users use Spark; I'd expect a guide on using Iceberg with Kedro to cover this.